saw-core/src/Verifier/SAW/Simulator/Value.hs

```diff
@@ -259,3 +259,28 @@ asFiniteTypeTValue v =
       FTRec <$> Map.fromList <$>
       mapM (\(fld,tp) -> (fld,) <$> asFiniteTypeTValue tp) elem_tps
     _ -> Nothing
+
+-- | A (partial) injective mapping from type values to strings. These
+-- are intended to be useful as suffixes for names of type instances
+-- of uninterpreted constants.
+suffixTValue :: TValue sym -> Maybe String
```

Comment:
Is this function for selecting type-based suffixes injective? I think it is, but it isn't immediately clear.

Reply:
Yes, it is. Basically it serializes the TValue with a pre-order traversal, and all the strings it uses to represent the nodes are distinct. So you could always unambiguously parse them back in.